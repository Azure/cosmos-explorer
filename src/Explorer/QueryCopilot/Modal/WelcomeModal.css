.modalContentPadding {
  padding-top: 15px;
  width: 513px;
  height: 638px;
}

.exitPadding {
  padding: 0px 7px 0px 0px;
}

.previewMargin {
  margin: 8px 10px 0px 0;
}

.preview {
  padding: 0px 4px;
  background: #f0f0f0;
  border-radius: 8px;
  font-size: 10px;
}

.exitIcon {
  margin: 3px 7px 0px 0px;
  color: #424242;
}

.text {
  width: 348px;
  padding: 8px 16px 8px 16px;
  margin-left: 25px;
}

.bold {
  font-weight: 600;
}

.imageTextPadding {
  padding: 0px 5px 0px 0px;
}

.buttonPadding {
  padding: 15px 0px 0px 0px;
}

.tryButton {
  border-radius: 4px;
}
