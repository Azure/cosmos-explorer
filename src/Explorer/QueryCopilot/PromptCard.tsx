import { ChoiceGroup, Stack, Text } from "@fluentui/react";
import React from "react";

interface PromptCardProps {
  header: string;
  description: string;
  isSelected: boolean;
  onSelect: () => void;
}

export const PromptCard: React.FC<PromptCardProps> = ({
  header,
  description,
  isSelected,
  onSelect,
}: PromptCardProps): JSX.Element => {
  return (
    <Stack
      horizontal
      style={{
        padding: "16px 0 16px 16px ",
        boxSizing: "border-box",
        width: 650,
        height: 100,
        border: "1px solid #F3F2F1",
        boxShadow: "0px 1.6px 3.6px rgba(0, 0, 0, 0.132), 0px 0.3px 0.9px rgba(0, 0, 0, 0.108)",
      }}
    >
      <Stack.Item grow={1}>
        <Stack horizontal>
          <div>
            <Text style={{ fontSize: 13, color: "#00A2AD", background: "#F8FFF0" }}>Prompt</Text>
          </div>
          <Text style={{ fontSize: 13, marginLeft: 16 }}>{header}</Text>
        </Stack>
        <Text style={{ fontSize: 10, marginTop: 16 }}>{description}</Text>
      </Stack.Item>
      <Stack.Item style={{ marginLeft: 16 }}>
        <ChoiceGroup
          styles={{ flexContainer: { width: 36 } }}
          options={[{ key: "selected", text: "" }]}
          selectedKey={isSelected ? "selected" : ""}
          onChange={onSelect}
        />
      </Stack.Item>
    </Stack>
  );
};
