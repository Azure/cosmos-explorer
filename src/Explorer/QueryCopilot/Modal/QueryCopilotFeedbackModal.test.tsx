import { Checkbox, DefaultButton, IconButton, PrimaryButton, TextField } from "@fluentui/react";
import Explorer from "Explorer/Explorer";
import { QueryCopilotFeedbackModal } from "Explorer/QueryCopilot/Modal/QueryCopilotFeedbackModal";
import { useCopilotStore } from "Explorer/QueryCopilot/QueryCopilotContext";
import { SubmitFeedback } from "Explorer/QueryCopilot/Shared/QueryCopilotClient";
import { getUserEmail } from "Utils/UserUtils";
import { shallow } from "enzyme";
import React from "react";

jest.mock("Utils/UserUtils");
(getUserEmail as jest.Mock).mockResolvedValue("test@email.com");

jest.mock("Explorer/QueryCopilot/Shared/QueryCopilotClient");
SubmitFeedback as jest.Mock;

jest.mock("Explorer/QueryCopilot/QueryCopilotContext");
const mockUseCopilotStore = useCopilotStore as jest.Mock;
const mockReturnValue = {
  generatedQuery: "test query",
  userPrompt: "test prompt",
  likeQuery: false,
  showFeedbackModal: false,
  closeFeedbackModal: jest.fn,
  setHideFeedbackModalForLikedQueries: jest.fn,
};

describe("Query Copilot Feedback Modal snapshot test", () => {
  beforeEach(() => {
    mockUseCopilotStore.mockReturnValue(mockReturnValue);
    jest.clearAllMocks();
  });
  it("shoud render and match snapshot", () => {
    mockUseCopilotStore.mockReturnValue({
      ...mockReturnValue,
      showFeedbackModal: true,
    });
    const wrapper = shallow(
      <QueryCopilotFeedbackModal
        explorer={new Explorer()}
        databaseId="CopilotUserDb"
        containerId="CopilotUserContainer"
        mode="User"
      />,
    );

    expect(wrapper.props().isOpen).toBeTruthy();
    expect(wrapper).toMatchSnapshot();
  });

  it("should close on cancel click", () => {
    const wrapper = shallow(
      <QueryCopilotFeedbackModal
        explorer={new Explorer()}
        databaseId="CopilotUserDb"
        containerId="CopilotUserContainer"
        mode="User"
      />,
    );

    const cancelButton = wrapper.find(IconButton);
    cancelButton.simulate("click");
    wrapper.setProps({});

    expect(wrapper.props().isOpen).toBeFalsy();
    expect(wrapper).toMatchSnapshot();
  });

  it("should get user unput", () => {
    const wrapper = shallow(
      <QueryCopilotFeedbackModal
        explorer={new Explorer()}
        databaseId="CopilotUserDb"
        containerId="CopilotUserContainer"
        mode="User"
      />,
    );
    const testUserInput = "test user input";

    const userInput = wrapper.find(TextField).first();
    userInput.simulate("change", {}, testUserInput);

    expect(wrapper.find(TextField).first().props().value).toEqual(testUserInput);
    expect(wrapper).toMatchSnapshot();
  });

  it("should not render dont show again button", () => {
    const wrapper = shallow(
      <QueryCopilotFeedbackModal
        explorer={new Explorer()}
        databaseId="CopilotUserDb"
        containerId="CopilotUserContainer"
        mode="User"
      />,
    );

    const dontShowAgain = wrapper.find(Checkbox);

    expect(dontShowAgain).toHaveLength(0);
    expect(wrapper).toMatchSnapshot();
  });

  it("should render dont show again button and check it ", () => {
    mockUseCopilotStore.mockReturnValue({
      ...mockReturnValue,
      showFeedbackModal: true,
      likeQuery: true,
    });
    const wrapper = shallow(
      <QueryCopilotFeedbackModal
        explorer={new Explorer()}
        databaseId="CopilotUserDb"
        containerId="CopilotUserContainer"
        mode="User"
      />,
    );

    const dontShowAgain = wrapper.find(Checkbox);
    dontShowAgain.simulate("change", {}, true);

    expect(wrapper.find(Checkbox)).toHaveLength(1);
    expect(wrapper.find(Checkbox).first().props().checked).toBeTruthy();
    expect(wrapper).toMatchSnapshot();
  });

  it("should cancel submission", () => {
    const wrapper = shallow(
      <QueryCopilotFeedbackModal
        explorer={new Explorer()}
        databaseId="CopilotUserDb"
        containerId="CopilotUserContainer"
        mode="User"
      />,
    );

    const cancelButton = wrapper.find(DefaultButton);
    cancelButton.simulate("click");
    wrapper.setProps({});

    expect(wrapper.props().isOpen).toBeFalsy();
    expect(wrapper).toMatchSnapshot();
  });

  it("should not submit submission if required description field is null", () => {
    const explorer = new Explorer();
    const wrapper = shallow(
      <QueryCopilotFeedbackModal
        explorer={explorer}
        databaseId="CopilotUserDb"
        containerId="CopilotUserContainer"
        mode="User"
      />,
    );

    const submitButton = wrapper.find(PrimaryButton);
    submitButton.simulate("click");
    wrapper.setProps({});

    expect(SubmitFeedback).toHaveBeenCalledTimes(0);
  });

  it("should submit submission", () => {
    const explorer = new Explorer();
    const wrapper = shallow(
      <QueryCopilotFeedbackModal
        explorer={explorer}
        databaseId="CopilotUserDb"
        containerId="CopilotUserContainer"
        mode="User"
      />,
    );

    const submitButton = wrapper.find("form");
    submitButton.simulate("submit");
    wrapper.setProps({});

    expect(SubmitFeedback).toHaveBeenCalledTimes(1);
    expect(SubmitFeedback).toHaveBeenCalledWith({
      containerId: "CopilotUserContainer",
      databaseId: "CopilotUserDb",
      mode: "User",
      params: {
        likeQuery: false,
        generatedQuery: "test query",
        userPrompt: "test prompt",
        description: "",
      },
      explorer: explorer,
    });
    expect(wrapper.props().isOpen).toBeFalsy();
    expect(wrapper).toMatchSnapshot();
  });
});
