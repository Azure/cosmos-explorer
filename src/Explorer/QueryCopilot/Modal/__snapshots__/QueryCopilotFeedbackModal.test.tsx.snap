// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Query Copilot Feedback Modal snapshot test shoud render and match snapshot 1`] = `
<Modal
  isOpen={true}
>
  <Stack
    style={
      Object {
        "padding": 24,
      }
    }
  >
    <Stack
      horizontal={true}
      horizontalAlign="space-between"
    >
      <Text
        style={
          Object {
            "fontSize": 20,
            "fontWeight": 600,
            "marginBottom": 20,
          }
        }
      >
        Send feedback to Microsoft
      </Text>
      <CustomizedIconButton
        iconProps={
          Object {
            "iconName": "Cancel",
          }
        }
        onClick={[Function]}
      />
    </Stack>
    <Text
      style={
        Object {
          "fontSize": 14,
          "marginBottom": 14,
        }
      }
    >
      Your feedback will help improve the experience.
    </Text>
    <StyledTextFieldBase
      label="Description"
      multiline={true}
      onChange={[Function]}
      placeholder="Provide more details"
      required={true}
      rows={3}
      styles={
        Object {
          "root": Object {
            "marginBottom": 14,
          },
        }
      }
      value=""
    />
    <StyledTextFieldBase
      defaultValue="test query"
      label="Query generated"
      readOnly={true}
      styles={
        Object {
          "root": Object {
            "marginBottom": 14,
          },
        }
      }
    />
    <StyledChoiceGroup
      label="May we contact you about your feedback?"
      onChange={[Function]}
      options={
        Array [
          Object {
            "key": "yes",
            "text": "Yes, you may contact me.",
          },
          Object {
            "key": "no",
            "text": "No, do not contact me.",
          },
        ]
      }
      selectedKey="yes"
      styles={
        Object {
          "flexContainer": Object {
            "selectors": Object {
              ".ms-ChoiceField-field::after": Object {
                "marginTop": 4,
              },
              ".ms-ChoiceField-field::before": Object {
                "marginTop": 4,
              },
              ".ms-ChoiceFieldLabel": Object {
                "paddingLeft": 6,
              },
            },
          },
          "root": Object {
            "marginBottom": 14,
          },
        }
      }
    />
    <Text
      style={
        Object {
          "fontSize": 12,
          "marginBottom": 14,
        }
      }
    >
      By pressing submit, your feedback will be used to improve Microsoft products and services. IT admins for your organization will be able to view and manage your feedback data.
       
      <StyledLinkBase
        href=""
        target="_blank"
      >
        Privacy statement
      </StyledLinkBase>
    </Text>
    <Stack
      horizontal={true}
      horizontalAlign="end"
    >
      <CustomizedPrimaryButton
        onClick={[Function]}
        styles={
          Object {
            "root": Object {
              "marginRight": 8,
            },
          }
        }
      >
        Submit
      </CustomizedPrimaryButton>
      <CustomizedDefaultButton
        onClick={[Function]}
      >
        Cancel
      </CustomizedDefaultButton>
    </Stack>
  </Stack>
</Modal>
`;

exports[`Query Copilot Feedback Modal snapshot test should cancel submission 1`] = `
<Modal
  isOpen={false}
>
  <Stack
    style={
      Object {
        "padding": 24,
      }
    }
  >
    <Stack
      horizontal={true}
      horizontalAlign="space-between"
    >
      <Text
        style={
          Object {
            "fontSize": 20,
            "fontWeight": 600,
            "marginBottom": 20,
          }
        }
      >
        Send feedback to Microsoft
      </Text>
      <CustomizedIconButton
        iconProps={
          Object {
            "iconName": "Cancel",
          }
        }
        onClick={[Function]}
      />
    </Stack>
    <Text
      style={
        Object {
          "fontSize": 14,
          "marginBottom": 14,
        }
      }
    >
      Your feedback will help improve the experience.
    </Text>
    <StyledTextFieldBase
      label="Description"
      multiline={true}
      onChange={[Function]}
      placeholder="Provide more details"
      required={true}
      rows={3}
      styles={
        Object {
          "root": Object {
            "marginBottom": 14,
          },
        }
      }
      value=""
    />
    <StyledTextFieldBase
      defaultValue=""
      label="Query generated"
      readOnly={true}
      styles={
        Object {
          "root": Object {
            "marginBottom": 14,
          },
        }
      }
    />
    <StyledChoiceGroup
      label="May we contact you about your feedback?"
      onChange={[Function]}
      options={
        Array [
          Object {
            "key": "yes",
            "text": "Yes, you may contact me.",
          },
          Object {
            "key": "no",
            "text": "No, do not contact me.",
          },
        ]
      }
      selectedKey="yes"
      styles={
        Object {
          "flexContainer": Object {
            "selectors": Object {
              ".ms-ChoiceField-field::after": Object {
                "marginTop": 4,
              },
              ".ms-ChoiceField-field::before": Object {
                "marginTop": 4,
              },
              ".ms-ChoiceFieldLabel": Object {
                "paddingLeft": 6,
              },
            },
          },
          "root": Object {
            "marginBottom": 14,
          },
        }
      }
    />
    <Text
      style={
        Object {
          "fontSize": 12,
          "marginBottom": 14,
        }
      }
    >
      By pressing submit, your feedback will be used to improve Microsoft products and services. IT admins for your organization will be able to view and manage your feedback data.
       
      <StyledLinkBase
        href=""
        target="_blank"
      >
        Privacy statement
      </StyledLinkBase>
    </Text>
    <Stack
      horizontal={true}
      horizontalAlign="end"
    >
      <CustomizedPrimaryButton
        onClick={[Function]}
        styles={
          Object {
            "root": Object {
              "marginRight": 8,
            },
          }
        }
      >
        Submit
      </CustomizedPrimaryButton>
      <CustomizedDefaultButton
        onClick={[Function]}
      >
        Cancel
      </CustomizedDefaultButton>
    </Stack>
  </Stack>
</Modal>
`;

exports[`Query Copilot Feedback Modal snapshot test should close on cancel click 1`] = `
<Modal
  isOpen={false}
>
  <Stack
    style={
      Object {
        "padding": 24,
      }
    }
  >
    <Stack
      horizontal={true}
      horizontalAlign="space-between"
    >
      <Text
        style={
          Object {
            "fontSize": 20,
            "fontWeight": 600,
            "marginBottom": 20,
          }
        }
      >
        Send feedback to Microsoft
      </Text>
      <CustomizedIconButton
        iconProps={
          Object {
            "iconName": "Cancel",
          }
        }
        onClick={[Function]}
      />
    </Stack>
    <Text
      style={
        Object {
          "fontSize": 14,
          "marginBottom": 14,
        }
      }
    >
      Your feedback will help improve the experience.
    </Text>
    <StyledTextFieldBase
      label="Description"
      multiline={true}
      onChange={[Function]}
      placeholder="Provide more details"
      required={true}
      rows={3}
      styles={
        Object {
          "root": Object {
            "marginBottom": 14,
          },
        }
      }
      value=""
    />
    <StyledTextFieldBase
      defaultValue=""
      label="Query generated"
      readOnly={true}
      styles={
        Object {
          "root": Object {
            "marginBottom": 14,
          },
        }
      }
    />
    <StyledChoiceGroup
      label="May we contact you about your feedback?"
      onChange={[Function]}
      options={
        Array [
          Object {
            "key": "yes",
            "text": "Yes, you may contact me.",
          },
          Object {
            "key": "no",
            "text": "No, do not contact me.",
          },
        ]
      }
      selectedKey="yes"
      styles={
        Object {
          "flexContainer": Object {
            "selectors": Object {
              ".ms-ChoiceField-field::after": Object {
                "marginTop": 4,
              },
              ".ms-ChoiceField-field::before": Object {
                "marginTop": 4,
              },
              ".ms-ChoiceFieldLabel": Object {
                "paddingLeft": 6,
              },
            },
          },
          "root": Object {
            "marginBottom": 14,
          },
        }
      }
    />
    <Text
      style={
        Object {
          "fontSize": 12,
          "marginBottom": 14,
        }
      }
    >
      By pressing submit, your feedback will be used to improve Microsoft products and services. IT admins for your organization will be able to view and manage your feedback data.
       
      <StyledLinkBase
        href=""
        target="_blank"
      >
        Privacy statement
      </StyledLinkBase>
    </Text>
    <Stack
      horizontal={true}
      horizontalAlign="end"
    >
      <CustomizedPrimaryButton
        onClick={[Function]}
        styles={
          Object {
            "root": Object {
              "marginRight": 8,
            },
          }
        }
      >
        Submit
      </CustomizedPrimaryButton>
      <CustomizedDefaultButton
        onClick={[Function]}
      >
        Cancel
      </CustomizedDefaultButton>
    </Stack>
  </Stack>
</Modal>
`;

exports[`Query Copilot Feedback Modal snapshot test should get user unput 1`] = `
<Modal
  isOpen={false}
>
  <Stack
    style={
      Object {
        "padding": 24,
      }
    }
  >
    <Stack
      horizontal={true}
      horizontalAlign="space-between"
    >
      <Text
        style={
          Object {
            "fontSize": 20,
            "fontWeight": 600,
            "marginBottom": 20,
          }
        }
      >
        Send feedback to Microsoft
      </Text>
      <CustomizedIconButton
        iconProps={
          Object {
            "iconName": "Cancel",
          }
        }
        onClick={[Function]}
      />
    </Stack>
    <Text
      style={
        Object {
          "fontSize": 14,
          "marginBottom": 14,
        }
      }
    >
      Your feedback will help improve the experience.
    </Text>
    <StyledTextFieldBase
      label="Description"
      multiline={true}
      onChange={[Function]}
      placeholder="Provide more details"
      required={true}
      rows={3}
      styles={
        Object {
          "root": Object {
            "marginBottom": 14,
          },
        }
      }
      value="test user input"
    />
    <StyledTextFieldBase
      defaultValue=""
      label="Query generated"
      readOnly={true}
      styles={
        Object {
          "root": Object {
            "marginBottom": 14,
          },
        }
      }
    />
    <StyledChoiceGroup
      label="May we contact you about your feedback?"
      onChange={[Function]}
      options={
        Array [
          Object {
            "key": "yes",
            "text": "Yes, you may contact me.",
          },
          Object {
            "key": "no",
            "text": "No, do not contact me.",
          },
        ]
      }
      selectedKey="yes"
      styles={
        Object {
          "flexContainer": Object {
            "selectors": Object {
              ".ms-ChoiceField-field::after": Object {
                "marginTop": 4,
              },
              ".ms-ChoiceField-field::before": Object {
                "marginTop": 4,
              },
              ".ms-ChoiceFieldLabel": Object {
                "paddingLeft": 6,
              },
            },
          },
          "root": Object {
            "marginBottom": 14,
          },
        }
      }
    />
    <Text
      style={
        Object {
          "fontSize": 12,
          "marginBottom": 14,
        }
      }
    >
      By pressing submit, your feedback will be used to improve Microsoft products and services. IT admins for your organization will be able to view and manage your feedback data.
       
      <StyledLinkBase
        href=""
        target="_blank"
      >
        Privacy statement
      </StyledLinkBase>
    </Text>
    <Stack
      horizontal={true}
      horizontalAlign="end"
    >
      <CustomizedPrimaryButton
        onClick={[Function]}
        styles={
          Object {
            "root": Object {
              "marginRight": 8,
            },
          }
        }
      >
        Submit
      </CustomizedPrimaryButton>
      <CustomizedDefaultButton
        onClick={[Function]}
      >
        Cancel
      </CustomizedDefaultButton>
    </Stack>
  </Stack>
</Modal>
`;

exports[`Query Copilot Feedback Modal snapshot test should not render dont show again button 1`] = `
<Modal
  isOpen={false}
>
  <Stack
    style={
      Object {
        "padding": 24,
      }
    }
  >
    <Stack
      horizontal={true}
      horizontalAlign="space-between"
    >
      <Text
        style={
          Object {
            "fontSize": 20,
            "fontWeight": 600,
            "marginBottom": 20,
          }
        }
      >
        Send feedback to Microsoft
      </Text>
      <CustomizedIconButton
        iconProps={
          Object {
            "iconName": "Cancel",
          }
        }
        onClick={[Function]}
      />
    </Stack>
    <Text
      style={
        Object {
          "fontSize": 14,
          "marginBottom": 14,
        }
      }
    >
      Your feedback will help improve the experience.
    </Text>
    <StyledTextFieldBase
      label="Description"
      multiline={true}
      onChange={[Function]}
      placeholder="Provide more details"
      required={true}
      rows={3}
      styles={
        Object {
          "root": Object {
            "marginBottom": 14,
          },
        }
      }
      value=""
    />
    <StyledTextFieldBase
      defaultValue=""
      label="Query generated"
      readOnly={true}
      styles={
        Object {
          "root": Object {
            "marginBottom": 14,
          },
        }
      }
    />
    <StyledChoiceGroup
      label="May we contact you about your feedback?"
      onChange={[Function]}
      options={
        Array [
          Object {
            "key": "yes",
            "text": "Yes, you may contact me.",
          },
          Object {
            "key": "no",
            "text": "No, do not contact me.",
          },
        ]
      }
      selectedKey="yes"
      styles={
        Object {
          "flexContainer": Object {
            "selectors": Object {
              ".ms-ChoiceField-field::after": Object {
                "marginTop": 4,
              },
              ".ms-ChoiceField-field::before": Object {
                "marginTop": 4,
              },
              ".ms-ChoiceFieldLabel": Object {
                "paddingLeft": 6,
              },
            },
          },
          "root": Object {
            "marginBottom": 14,
          },
        }
      }
    />
    <Text
      style={
        Object {
          "fontSize": 12,
          "marginBottom": 14,
        }
      }
    >
      By pressing submit, your feedback will be used to improve Microsoft products and services. IT admins for your organization will be able to view and manage your feedback data.
       
      <StyledLinkBase
        href=""
        target="_blank"
      >
        Privacy statement
      </StyledLinkBase>
    </Text>
    <Stack
      horizontal={true}
      horizontalAlign="end"
    >
      <CustomizedPrimaryButton
        onClick={[Function]}
        styles={
          Object {
            "root": Object {
              "marginRight": 8,
            },
          }
        }
      >
        Submit
      </CustomizedPrimaryButton>
      <CustomizedDefaultButton
        onClick={[Function]}
      >
        Cancel
      </CustomizedDefaultButton>
    </Stack>
  </Stack>
</Modal>
`;

exports[`Query Copilot Feedback Modal snapshot test should record user contact choice no 1`] = `
<Modal
  isOpen={false}
>
  <Stack
    style={
      Object {
        "padding": 24,
      }
    }
  >
    <Stack
      horizontal={true}
      horizontalAlign="space-between"
    >
      <Text
        style={
          Object {
            "fontSize": 20,
            "fontWeight": 600,
            "marginBottom": 20,
          }
        }
      >
        Send feedback to Microsoft
      </Text>
      <CustomizedIconButton
        iconProps={
          Object {
            "iconName": "Cancel",
          }
        }
        onClick={[Function]}
      />
    </Stack>
    <Text
      style={
        Object {
          "fontSize": 14,
          "marginBottom": 14,
        }
      }
    >
      Your feedback will help improve the experience.
    </Text>
    <StyledTextFieldBase
      label="Description"
      multiline={true}
      onChange={[Function]}
      placeholder="Provide more details"
      required={true}
      rows={3}
      styles={
        Object {
          "root": Object {
            "marginBottom": 14,
          },
        }
      }
      value=""
    />
    <StyledTextFieldBase
      defaultValue=""
      label="Query generated"
      readOnly={true}
      styles={
        Object {
          "root": Object {
            "marginBottom": 14,
          },
        }
      }
    />
    <StyledChoiceGroup
      label="May we contact you about your feedback?"
      onChange={[Function]}
      options={
        Array [
          Object {
            "key": "yes",
            "text": "Yes, you may contact me.",
          },
          Object {
            "key": "no",
            "text": "No, do not contact me.",
          },
        ]
      }
      selectedKey="no"
      styles={
        Object {
          "flexContainer": Object {
            "selectors": Object {
              ".ms-ChoiceField-field::after": Object {
                "marginTop": 4,
              },
              ".ms-ChoiceField-field::before": Object {
                "marginTop": 4,
              },
              ".ms-ChoiceFieldLabel": Object {
                "paddingLeft": 6,
              },
            },
          },
          "root": Object {
            "marginBottom": 14,
          },
        }
      }
    />
    <Text
      style={
        Object {
          "fontSize": 12,
          "marginBottom": 14,
        }
      }
    >
      By pressing submit, your feedback will be used to improve Microsoft products and services. IT admins for your organization will be able to view and manage your feedback data.
       
      <StyledLinkBase
        href=""
        target="_blank"
      >
        Privacy statement
      </StyledLinkBase>
    </Text>
    <Stack
      horizontal={true}
      horizontalAlign="end"
    >
      <CustomizedPrimaryButton
        onClick={[Function]}
        styles={
          Object {
            "root": Object {
              "marginRight": 8,
            },
          }
        }
      >
        Submit
      </CustomizedPrimaryButton>
      <CustomizedDefaultButton
        onClick={[Function]}
      >
        Cancel
      </CustomizedDefaultButton>
    </Stack>
  </Stack>
</Modal>
`;

exports[`Query Copilot Feedback Modal snapshot test should record user contact choice yes 1`] = `
<Modal
  isOpen={false}
>
  <Stack
    style={
      Object {
        "padding": 24,
      }
    }
  >
    <Stack
      horizontal={true}
      horizontalAlign="space-between"
    >
      <Text
        style={
          Object {
            "fontSize": 20,
            "fontWeight": 600,
            "marginBottom": 20,
          }
        }
      >
        Send feedback to Microsoft
      </Text>
      <CustomizedIconButton
        iconProps={
          Object {
            "iconName": "Cancel",
          }
        }
        onClick={[Function]}
      />
    </Stack>
    <Text
      style={
        Object {
          "fontSize": 14,
          "marginBottom": 14,
        }
      }
    >
      Your feedback will help improve the experience.
    </Text>
    <StyledTextFieldBase
      label="Description"
      multiline={true}
      onChange={[Function]}
      placeholder="Provide more details"
      required={true}
      rows={3}
      styles={
        Object {
          "root": Object {
            "marginBottom": 14,
          },
        }
      }
      value=""
    />
    <StyledTextFieldBase
      defaultValue=""
      label="Query generated"
      readOnly={true}
      styles={
        Object {
          "root": Object {
            "marginBottom": 14,
          },
        }
      }
    />
    <StyledChoiceGroup
      label="May we contact you about your feedback?"
      onChange={[Function]}
      options={
        Array [
          Object {
            "key": "yes",
            "text": "Yes, you may contact me.",
          },
          Object {
            "key": "no",
            "text": "No, do not contact me.",
          },
        ]
      }
      selectedKey="yes"
      styles={
        Object {
          "flexContainer": Object {
            "selectors": Object {
              ".ms-ChoiceField-field::after": Object {
                "marginTop": 4,
              },
              ".ms-ChoiceField-field::before": Object {
                "marginTop": 4,
              },
              ".ms-ChoiceFieldLabel": Object {
                "paddingLeft": 6,
              },
            },
          },
          "root": Object {
            "marginBottom": 14,
          },
        }
      }
    />
    <Text
      style={
        Object {
          "fontSize": 12,
          "marginBottom": 14,
        }
      }
    >
      By pressing submit, your feedback will be used to improve Microsoft products and services. IT admins for your organization will be able to view and manage your feedback data.
       
      <StyledLinkBase
        href=""
        target="_blank"
      >
        Privacy statement
      </StyledLinkBase>
    </Text>
    <Stack
      horizontal={true}
      horizontalAlign="end"
    >
      <CustomizedPrimaryButton
        onClick={[Function]}
        styles={
          Object {
            "root": Object {
              "marginRight": 8,
            },
          }
        }
      >
        Submit
      </CustomizedPrimaryButton>
      <CustomizedDefaultButton
        onClick={[Function]}
      >
        Cancel
      </CustomizedDefaultButton>
    </Stack>
  </Stack>
</Modal>
`;

exports[`Query Copilot Feedback Modal snapshot test should render dont show again button and check it  1`] = `
<Modal
  isOpen={true}
>
  <Stack
    style={
      Object {
        "padding": 24,
      }
    }
  >
    <Stack
      horizontal={true}
      horizontalAlign="space-between"
    >
      <Text
        style={
          Object {
            "fontSize": 20,
            "fontWeight": 600,
            "marginBottom": 20,
          }
        }
      >
        Send feedback to Microsoft
      </Text>
      <CustomizedIconButton
        iconProps={
          Object {
            "iconName": "Cancel",
          }
        }
        onClick={[Function]}
      />
    </Stack>
    <Text
      style={
        Object {
          "fontSize": 14,
          "marginBottom": 14,
        }
      }
    >
      Your feedback will help improve the experience.
    </Text>
    <StyledTextFieldBase
      label="Description"
      multiline={true}
      onChange={[Function]}
      placeholder="Provide more details"
      required={true}
      rows={3}
      styles={
        Object {
          "root": Object {
            "marginBottom": 14,
          },
        }
      }
      value=""
    />
    <StyledTextFieldBase
      defaultValue="test query"
      label="Query generated"
      readOnly={true}
      styles={
        Object {
          "root": Object {
            "marginBottom": 14,
          },
        }
      }
    />
    <StyledChoiceGroup
      label="May we contact you about your feedback?"
      onChange={[Function]}
      options={
        Array [
          Object {
            "key": "yes",
            "text": "Yes, you may contact me.",
          },
          Object {
            "key": "no",
            "text": "No, do not contact me.",
          },
        ]
      }
      selectedKey="yes"
      styles={
        Object {
          "flexContainer": Object {
            "selectors": Object {
              ".ms-ChoiceField-field::after": Object {
                "marginTop": 4,
              },
              ".ms-ChoiceField-field::before": Object {
                "marginTop": 4,
              },
              ".ms-ChoiceFieldLabel": Object {
                "paddingLeft": 6,
              },
            },
          },
          "root": Object {
            "marginBottom": 14,
          },
        }
      }
    />
    <Text
      style={
        Object {
          "fontSize": 12,
          "marginBottom": 14,
        }
      }
    >
      By pressing submit, your feedback will be used to improve Microsoft products and services. IT admins for your organization will be able to view and manage your feedback data.
       
      <StyledLinkBase
        href=""
        target="_blank"
      >
        Privacy statement
      </StyledLinkBase>
    </Text>
    <StyledCheckboxBase
      checked={true}
      label="Don't show me this next time"
      onChange={[Function]}
      styles={
        Object {
          "label": Object {
            "paddingLeft": 0,
          },
          "root": Object {
            "marginBottom": 14,
          },
        }
      }
    />
    <Stack
      horizontal={true}
      horizontalAlign="end"
    >
      <CustomizedPrimaryButton
        onClick={[Function]}
        styles={
          Object {
            "root": Object {
              "marginRight": 8,
            },
          }
        }
      >
        Submit
      </CustomizedPrimaryButton>
      <CustomizedDefaultButton
        onClick={[Function]}
      >
        Cancel
      </CustomizedDefaultButton>
    </Stack>
  </Stack>
</Modal>
`;

exports[`Query Copilot Feedback Modal snapshot test should submit submission 1`] = `
<Modal
  isOpen={false}
>
  <Stack
    style={
      Object {
        "padding": 24,
      }
    }
  >
    <Stack
      horizontal={true}
      horizontalAlign="space-between"
    >
      <Text
        style={
          Object {
            "fontSize": 20,
            "fontWeight": 600,
            "marginBottom": 20,
          }
        }
      >
        Send feedback to Microsoft
      </Text>
      <CustomizedIconButton
        iconProps={
          Object {
            "iconName": "Cancel",
          }
        }
        onClick={[Function]}
      />
    </Stack>
    <Text
      style={
        Object {
          "fontSize": 14,
          "marginBottom": 14,
        }
      }
    >
      Your feedback will help improve the experience.
    </Text>
    <StyledTextFieldBase
      label="Description"
      multiline={true}
      onChange={[Function]}
      placeholder="Provide more details"
      required={true}
      rows={3}
      styles={
        Object {
          "root": Object {
            "marginBottom": 14,
          },
        }
      }
      value=""
    />
    <StyledTextFieldBase
      defaultValue=""
      label="Query generated"
      readOnly={true}
      styles={
        Object {
          "root": Object {
            "marginBottom": 14,
          },
        }
      }
    />
    <StyledChoiceGroup
      label="May we contact you about your feedback?"
      onChange={[Function]}
      options={
        Array [
          Object {
            "key": "yes",
            "text": "Yes, you may contact me.",
          },
          Object {
            "key": "no",
            "text": "No, do not contact me.",
          },
        ]
      }
      selectedKey="yes"
      styles={
        Object {
          "flexContainer": Object {
            "selectors": Object {
              ".ms-ChoiceField-field::after": Object {
                "marginTop": 4,
              },
              ".ms-ChoiceField-field::before": Object {
                "marginTop": 4,
              },
              ".ms-ChoiceFieldLabel": Object {
                "paddingLeft": 6,
              },
            },
          },
          "root": Object {
            "marginBottom": 14,
          },
        }
      }
    />
    <Text
      style={
        Object {
          "fontSize": 12,
          "marginBottom": 14,
        }
      }
    >
      By pressing submit, your feedback will be used to improve Microsoft products and services. IT admins for your organization will be able to view and manage your feedback data.
       
      <StyledLinkBase
        href=""
        target="_blank"
      >
        Privacy statement
      </StyledLinkBase>
    </Text>
    <Stack
      horizontal={true}
      horizontalAlign="end"
    >
      <CustomizedPrimaryButton
        onClick={[Function]}
        styles={
          Object {
            "root": Object {
              "marginRight": 8,
            },
          }
        }
      >
        Submit
      </CustomizedPrimaryButton>
      <CustomizedDefaultButton
        onClick={[Function]}
      >
        Cancel
      </CustomizedDefaultButton>
    </Stack>
  </Stack>
</Modal>
`;
